test/framework/self-tests/fail_assert2.c

```diff
@@ -0,0 +1,23 @@
+/**
+ * @file test/self-tests/fail_assert.c
+ *
+ * @brief Test: Test core functions of the testing framework
+ *
+ * This is the test that checks if assert can fail a test.
+ *
+ * SPDX-FileCopyrightText: 2008-2021 HPDCS Group <rootsim@googlegroups.com>
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+#include <test.h>
+#include <string.h>
+
+extern test_ret_t test_pass(__unused void *_);
+extern test_ret_t test_pass_assert(__unused void *_);
```

Comment:
If I am not mistaken, this declaration is not needed as well

Reply:
It's not. It just loved its childhood friend.